From e8db1868cf30c1e8c0ae85a68e7bfc424e0ceba7 Mon Sep 17 00:00:00 2001
From: Arnon Kaufman <akaufman@habana.ai>
Date: Tue, 22 Feb 2022 18:49:34 +0200
Subject: [PATCH] [SW-65638] gaudi2: release: fix argument parsing

Avoiding evaluation an illegal variable name.
An example: a configuration variable set dynamically with wrong format.
--@40, or --@4asd=t, will result with error. This patch comes to solve
it.

Change-Id: I48d4da06832a790f9ab495d10eb1be084dfeb9bd
Signed-off-by: Arnon Kaufman <akaufman@habana.ai>
---

diff --git a/scripts/fwi/fw_installer.sh b/scripts/fwi/fw_installer.sh
index a6beae4..17ee2b7 100755
--- a/scripts/fwi/fw_installer.sh
+++ b/scripts/fwi/fw_installer.sh
@@ -1344,8 +1344,13 @@
 				perr "Unknown config varible $cmd args [${args[@]}]"
 				exit 1
 			}
-			eval ${config_var}="${args[@]}"
-			config_vars_list+=($cmd)
+			[[ ${config_var} =~ ^[[:alpha:]] ]] && {
+				eval ${config_var}="${args[@]}"
+				config_vars_list+=($cmd)
+			} || {
+				perr "Illegal argument [${config_var}]"
+				exit 1
+			}
 		esac
 		shift $nargs
 	done
